source/aggregation.txt

```diff
@@ -0,0 +1,210 @@
+  value.
+
+.. io-code-block::
+
+   .. input:: /includes/aggregation/aggregation.kt
+      :language: kotlin
+      :start-after: start-aggregation-pipeline
+      :end-before: end-aggregation-pipeline
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      Document{{_id=Bronx, count=71}}
+      Document{{_id=Manhattan, count=221}}
+      Document{{_id=Brooklyn, count=173}}
+      Document{{_id=Queens, count=204}}
+      Document{{_id=Staten Island, count=20}}
+      Document{{_id=Missing, count=2}}
+
+Explain an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~
+
+To view information about how MongoDB executes your operation, you can
+instruct MongoDB to **explain** it. When MongoDB explains an operation, it returns
+**execution plans** and performance statistics. An execution
+plan is a potential way MongoDB can complete an operation.
+When you instruct MongoDB to explain an operation, it returns both the
+plan MongoDB executed and any rejected execution plans.
+
+The following code example runs the preceding aggregation example and prints the returned
+explanation:
+
+.. io-code-block::
+
+   .. input:: /includes/aggregation/aggregation.kt
+      :language: kotlin
+      :start-after: start-aggregation-explain
+      :end-before: end-aggregation-explain
+      :dedent:
+
+   .. output::
+      :visible: false
+    
+      {
+        "explain": {
+          "aggregate": "restaurants",
+          "pipeline": [
+            {
+              "$match": {
+                "cuisine": "Bakery"
+              }
+            },
+            {
+              "$group": {
+                "_id": "$borough",
+                "count": {
+                  "$sum": 1
+                }
+              }
+            }
+          ],
+          "cursor": {}
+        },
+        ...
+      }
+
+Additional Information
+----------------------
+
+To view a full list of expression operators, see :manual:`Aggregation
+Operators. </reference/operator/aggregation/>`
+
+To learn about assembling an aggregation pipeline and view examples, see
+:manual:`Aggregation Pipeline. </core/aggregation-pipeline/>`
+
+To learn more about creating pipeline stages, see :manual:`Aggregation
+Stages. </reference/operator/aggregation-pipeline/>`
+
+To learn more about explaining MongoDB operations, see
+:manual:`Explain Output </reference/explain-results/>` and
+:manual:`Query Plans. </core/query-plans/>`
+
+.. Aggregation Tutorials
+.. ~~~~~~~~~~~~~~~~~~~~~
+
+.. To view step-by-step explanations of common aggregation tasks, see
+.. :ref:`kotlin-sync-aggregation-tutorials`.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+For more information about executing aggregation operations with the {+driver-short+},
+see the following API documentation:
+
+- `aggregate() <{+api+}/com.mongodb.kotlin.client/-mongo-collection/aggregate.html>`__
+- `AggregateIterable <{+api+}/com.mongodb.kotlin.client/-aggregate-iterable/index.html>`__
```

Comment:
Note for reviewer: these links are broken at the moment.

Rea has a change that has yet to be merged that is changes the `{+api+}` source constant. This will work once it's merged.